### ISSUE TITLE
The default value of `interpolation.skipOnVariables` is falsy if other interpolation option is specified

### ISSUE THREAD

**Issue by ishikawa** [i18next/i18next#1711]
## 🐛 Bug Report

[According to the documentation](https://www.i18next.com/translation-function/interpolation#all-interpolation-options), the default value of `interpolation.skipOnVariables` is `true` since v21.0.0. However, even in v21.6.3, `interpolation.skipOnVariables` seems to be set to falsy if other interpolation options are specified.

## To Reproduce

Without any interporation options, the default values are provided by expected.

```typescript
import i18n from 'i18next';

i18n
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({ lng: 'en-US' });

console.log('>>> interpolation =', i18n.options.interpolation);
```

```
>>> interpolation = Object {
  "escapeValue": true,
  "format": [Function bound format],
  "formatSeparator": ",",
  "maxReplaces": 1000,
  "nestingOptionsSeparator": ",",
  "nestingPrefix": "$t(",
  "nestingSuffix": ")",
  "prefix": "{{",
  "skipOnVariables": true,
  "suffix": "}}",
  "unescapePrefix": "-",
}
```

But if other interporation option (I chose `escapeValue` option here) is specified, `skipOnVariables` is set to `undefined` and it doesn't seem to work.

```typescript
i18n
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    lng: 'en-US',
    interpolation: {
      escapeValue: false, // react already safes from xss => https://www.i18next.com/translation-function/interpolation#unescape
    },
  });

console.log('>>> interpolation =', i18n.options.interpolation);
```

```
>>> interpolation = Object {
  "escapeValue": false,
  "format": [Function bound format],
}
```

## Expected behavior

Regardless of other options, the default value of `skipOnVariables` should be `true`.

```js
i18n
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    lng: 'en-US',
    interpolation: {
      escapeValue: false, // react already safes from xss => https://www.i18next.com/translation-function/interpolation#unescape
    },
  });

console.log('>>> interpolation =', i18n.options.interpolation);
```

```
>>> interpolation = Object {
  "escapeValue": false,
  "skipOnVariables": true,
  "format": [Function bound format],
}
```

## Your Environment

- *runtime version*: 16.13.1
- *i18next version*: 21.6.3
- *os*: Mac
- *any other relevant information*

**Comment by adrai** [comment]
v21.6.4 should be better

**Comment by ishikawa** [comment]
@adrai Thanks for hot fix v21.6.4. I updated and checked the expected result. 😄 
```
>>> interpolation Object {
  "escapeValue": false,
  "format": [Function bound format],
  "formatSeparator": ",",
  "maxReplaces": 1000,
  "nestingOptionsSeparator": ",",
  "nestingPrefix": "$t(",
  "nestingSuffix": ")",
  "prefix": "{{",
  "skipOnVariables": true,
  "suffix": "}}",
  "unescapePrefix": "-",
}
```

**Comment by mrlubos** [comment]
Hey @adrai @ishikawa, v21.6.4 might've fixed one issue, but it introduced another problem, see https://github.com/i18next/i18next/issues/1721 please
